unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/Debugging/FPSDisplay/FPSDisplay.cs

```diff
@@ -24,22 +29,111 @@ public class FPSDisplay : MonoBehaviour
+        }
 
+        private void OnEnable()
+        {
             lastPlayerCount = otherPlayers.Count();
             otherPlayers.OnAdded += OnOtherPlayersModified;
             otherPlayers.OnRemoved += OnOtherPlayersModified;
 
             SetupKernelConfig();
             SetupRealm();
 
+            SetupDictionary();
+            
+            ProfilingEvents.OnMessageWillQueue += OnMessageWillQueue;
+            ProfilingEvents.OnMessageWillDequeue += OnMessageWillDequeue;
+
             StartCoroutine(UpdateLabelLoop());
         }
+        
+        private void OnDisable()
+        {
+            otherPlayers.OnAdded -= OnOtherPlayersModified;
+            otherPlayers.OnRemoved -= OnOtherPlayersModified;
+            currentRealm.OnChange -= UpdateRealm;
+            kernelConfigPromise.Dispose();
+            KernelConfig.i.OnChange -= OnKernelConfigChanged;
+            
+            ProfilingEvents.OnMessageWillQueue += OnMessageWillQueue;
+            ProfilingEvents.OnMessageWillDequeue += OnMessageWillDequeue;
 
+            StopAllCoroutines();
+        }
+
+        private void SetupDictionary()
+        {
+            if (updateValueDictionary != null)
+                return;
+            
+            updateValueDictionary = new Dictionary<DebugValueEnum, Func<string>>();
+            updateValueDictionary.Add(DebugValueEnum.Skybox_Config, () => DataStore.i.skyboxConfig.configToLoad.Get());
+            updateValueDictionary.Add(DebugValueEnum.Skybox_Duration, () => $"{DataStore.i.skyboxConfig.lifecycleDuration.Get()}");
+            updateValueDictionary.Add(DebugValueEnum.Skybox_GameTime, () => $"{DataStore.i.skyboxConfig.currentVirtualTime.Get().ToString(TWO_DECIMALS)}");
+            updateValueDictionary.Add(DebugValueEnum.Skybox_UTCTime, () => $"{DataStore.i.worldTimer.GetCurrentTime()}");
+            
+            updateValueDictionary.Add(DebugValueEnum.General_Network, () => currentNetwork.ToUpper());
+            updateValueDictionary.Add(DebugValueEnum.General_Realm, () => currentRealmValue.ToUpper());
+            updateValueDictionary.Add(DebugValueEnum.General_NearbyPlayers, () => lastPlayerCount.ToString());
+            
+            updateValueDictionary.Add(DebugValueEnum.FPS, GetFPSCount);
+            updateValueDictionary.Add(DebugValueEnum.FPS_HiccupsInTheLast1000, () => $"{fpsColor}{performanceData.Get().hiccupCount.ToString()}</color>");
+            updateValueDictionary.Add(DebugValueEnum.FPS_HiccupsLoss, GetHiccupsLoss);
+            updateValueDictionary.Add(DebugValueEnum.FPS_BadFramesPercentiles, () => $"{fpsColor}{((performanceData.Get().hiccupCount) / 10.0f).ToString(NO_DECIMALS)}%</color>");
+            
+            updateValueDictionary.Add(DebugValueEnum.Scene_Name, () =>  GetStringID());
+            updateValueDictionary.Add(DebugValueEnum.Scene_ProcessedMessages, () =>  $"{totalMessagesCurrent} of {totalMessagesGlobal}");
```

Comment:
I would change this to:

`%{totalMessagesCurrent / totalMessagesGlobal * 100}`

 since those big numbers wont give us much information, if that number falls from 100 then we have a problem.